chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RealWorldTestingEligibilityJob.java

```diff
@@ -0,0 +1,200 @@
+            throw new RuntimeException("Could not retrieve details for listings.", e);
+        }
+
+    }
+
+    private boolean doesListingAttestToEligibleCriteria(
+            CertifiedProductSearchDetails listing, List<CertificationCriterion> eligibleCriteria) {
+        boolean doesExist = listing.getCertificationResults().stream()
+                .filter(result -> result.isSuccess()
+                        && eligibleCriteria.stream()
+                        .filter(crit -> crit.getId().equals(result.getCriterion().getId()))
+                        .findAny()
+                        .isPresent())
+                .findAny()
+                .isPresent();
+
+        if (doesExist) {
+            return true;
+        } else {
+            LOGGER.info("Listing: " + listing.getId() + " - Does not attest to any eligible criteria");
+            return false;
+        }
+    }
+
+    private CertifiedProductSearchDetails getCertifiedProductSearchDetails(Long listingId) {
+        try {
+            CertifiedProductSearchDetails detail = certifiedProductDetailsManager.getCertifiedProductDetails(listingId);
+            return detail;
+        } catch (Exception e) {
+            LOGGER.error("Could not retrieve the details for listing: " + listingId);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private List<CertificationCriterion> getRwtEligibleCriteria() {
+        return Arrays.asList(eligibleCriteriaKeys).stream()
+                .map(key -> certificationCriterionService.get(key))
+                .collect(Collectors.toList());
+    }
+    private List<CertifiedProductDetailsDTO> getAllListingsWith2015Edition() {
+        LOGGER.info("Getting all 2015 listings.");
+        List<CertifiedProductDetailsDTO> listings = certifiedProductDAO.findByEdition(
+                CertificationEditionConcept.CERTIFICATION_EDITION_2015.getYear());
+        LOGGER.info("Completing getting all 2015 listings. Found " + listings.size() + " listings.");
+        return listings;
+    }
+
+    private boolean doesListingHaveExistingRwtEligibility(CertifiedProductDetailsDTO listing) {
+        if (Objects.nonNull(listing.getRwtEligibilityYear())) {
+            LOGGER.info("Listing: " + listing.getId() + " - Already RWT eligible");
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean isListingStatusActiveAsOfDate(Long listingId, Date asOfDate) {
+        CertificationStatusEventDTO event = getListingStatusAsOf(listingId, asOfDate);
+        if (Objects.nonNull(event)
+                && event.getStatus().getStatus().equals(RealWorldTestingEligibilityJob.ACTIVE)) {
+            return true;
+        } else {
+            LOGGER.info("Listing: " + listingId + " - Not Active");
+            return false;
+        }
+    }
+
+    private CertificationStatusEventDTO getListingStatusAsOf(Long listingId, Date asOfDate) {
```

Comment:
There is a method in CertifiedProductSearchDetails called getStatusOnDate because I've had to do this before. Could you make use of it here?